### ISSUE TITLE
`unneeded_throws_rethrows` shows false positives for protocol conformances

### ISSUE THREAD

**Issue by username0x0a** [realm/SwiftLint#6437]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

When opting in to a `unneeded_throws_rethrows`, I'm getting false positives for the following case:

```swift
protocol Throwing {
    func doSomething() throws
}

struct SomeType: Throwing {

                        vvvv
    func doSomething() throws {
                        ^^^^
        // Closure without anything thrown
        print("Hello")
    }
}

```

I believe that this case where the `throws` declaration is **enforced** by the protocol conformance shouldn't trigger a positive report for this rule.

### Environment

* SwiftLint 0.63.0
* Xcode 26.2 (17C52)
* Installed via Homebrew

```yml
...
opt_in_rules:
- unneeded_throws_rethrows
...
```

**Comment by SimplyDanny** [comment]
This example works for me without the `throws` keyword in the struct. Generally, this is allowed. To conform to a throwing function, the implementor doesn't need to be throwing. Reason: The set of non-throwing functions is a subset of the set of throwing functions.
